documents/anchor_task.go

```diff
@@ -119,31 +119,41 @@ func (d *documentAnchorTask) RunTask() (res interface{}, err error) {
 	return true, nil
 }
 
-// InitDocumentAnchorTask enqueues a new document anchor task and returns the txID.
-// TODO [TXManager] migrate this to use TxManager
-func InitDocumentAnchorTask(tq queue.TaskQueuer, txService transactions.Manager, accountID identity.CentID, modelID []byte, txID uuid.UUID) (uuid.UUID, error) {
-	var tx *transactions.Transaction
-	var err error
-	if txID != uuid.Nil {
-		tx, err = txService.GetTransaction(accountID, txID)
-	} else {
-		tx, err = txService.CreateTransaction(accountID, documentAnchorTaskName)
-	}
-
-	if err != nil {
-		return uuid.Nil, err
-	}
-
+// InitDocumentAnchorTask enqueues a new document anchor task for a given combination of accountID/modelID/txID.
+func InitDocumentAnchorTask(txMan transactions.Manager, tq queue.TaskQueuer, accountID identity.CentID, modelID []byte, txID uuid.UUID) (queue.TaskResult, error) {
 	params := map[string]interface{}{
-		transactions.TxIDParam: tx.ID.String(),
+		transactions.TxIDParam: txID.String(),
 		DocumentIDParam:        hexutil.Encode(modelID),
 		AccountIDParam:         accountID.String(),
 	}
 
-	_, err = tq.EnqueueJob(documentAnchorTaskName, params)
+	err := txMan.UpdateTaskStatus(accountID, txID, transactions.Pending, documentAnchorTaskName, "init")
 	if err != nil {
-		return uuid.Nil, err
+		return nil, err
 	}
 
-	return tx.ID, nil
+	tr, err := tq.EnqueueJob(documentAnchorTaskName, params)
+	if err != nil {
+		return nil, err
+	}
+
+	return tr, nil
+}
+
+// CreateAnchorTransaction creates a transaction for anchoring a document using transaction manager
+func CreateAnchorTransaction(txMan transactions.Manager, tq queue.TaskQueuer, self identity.CentID, txID uuid.UUID, documentID []byte) (uuid.UUID, chan bool, error) {
```

Comment:
this would be used by all doc updates to create or update an existing tx with task status.

Reply:
need to add task status as pending here